File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/proxy/simple/Proxy.java
// Proxy.java

package com.bigboxcode.designpattern.proxy.simple;

public class Proxy implements ProxyInterface {
    private ProxyInterface actual;

    @Override
    public void operation1() {
         if (actual == null) {
             actual = new Actual();
         }

        actual.operation1();
    }

    @Override
    public void operation2() {
        if (actual == null) {
            actual = new Actual();
        }

        actual.operation2();
    }
}

File: design-patterns/php/proxy/simpleproxy/Proxy.php
<?php
// Proxy.php

namespace BigBoxCode\DesignPattern\Proxy\SimpleProxy;

class Proxy implements ProxyInterface {
    private ?ProxyInterface $item = null;

    public function operation1(): void {
        if ($this->item == null) {
            $this->item = new Item();
        }

        $this->item->operation1();
    }

    public function operation2(): void {
        if ($this->item == null) {
            $this->item = new Item();
        }

        $this->item->operation2();
    }
}